### PR TITLE
Configuration management Part 2: Help Links + Email Templates

### PR DESCRIPTION
**Pull request by steve-gregory**
This feature, ported from Troposphere, allows the application and the GUI to behave in a coordinated fashion.

New features include:

- [x] Email Template now has many 'links' 
- [x] HelpLink is created in the same migration as template
- [x] Data migration included in the same file.
- [x] Reorganized migrations after m-m merge conflicts.
- [x] E-mail now provides users with more information and more help.
- [x] E-mail now configurable by "Site Operators" via database

### CODE REVIEW

**Comment by steve-gregory** [comment]
@lenards to you for final review.

File: core/admin.py [hunk 1]
```diff
@@ -528,9 +531,35 @@ class GroupAdmin(admin.ModelAdmin):
...
+        if obj: # editing an existing object
+            return self.readonly_fields + ("link_key", )
+        return self.readonly_fields
+
+    def has_add_permission(self, request):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
```
Comment (lenards) [inline core/admin.py]: :+1:

File: core/models/__init__.py [hunk 2]
```diff
@@ -22,8 +22,11 @@
 from core.models.project_link import ProjectExternalLink
 from core.models.project_instance import ProjectInstance
 from core.models.project_volume import ProjectVolume
-from core.models.provider import AccountProvider, ProviderType, PlatformType,\
-    Provider, ProviderInstanceAction, ProviderDNSServerIP
+from core.models.provider import (
+    AccountProvider, ProviderType, PlatformType,
+    Provider, ProviderInstanceAction, ProviderDNSServerIP,
+    ProviderConfiguration
+)
```
Comment (lenards) [inline core/models/__init__.py]: :100:

File: core/models/status_type.py [hunk 3]
```diff
@@ -47,4 +47,5 @@ def __unicode__(self):
             (self.name,)
 
     def __str__(self):
-        return self.__unicode__()
+        return "%s" %\
```
Comment (lenards) [inline core/models/status_type.py]: I never know if you should have a space between `%` & `\` or not.

File: core/models/template.py [hunk 4]
```diff
@@ -20,13 +20,42 @@ class EmailTemplate(SingletonModel):
...
+    def get_link(self, link_key):
+        try:
+            return self.links.get(link_key=link_key)
+        except models.ObjectDoesNotExist:
+            return None
+
 
     class Meta:
         db_table = 'email_template'
         app_label = 'core'
```
Comment (lenards) [inline core/models/template.py]: 2 spaces between the `Meta` and the last method?

File: variables.ini.dist [hunk 5]
```diff
@@ -84,6 +84,7 @@ TEST_RUNNER_PASS = ; test_secret
 MOCK_USER = ; test_user
 USE_MOCK_AUTH = ; False
 #local.py - site configuration
+TIME_ZONE = ; America/Phoenix
```
Comment (lenards) [inline variables.ini.dist]: :clock1:

**Comment by lenards** [comment]
Works for me! 

:shipit:

File: core/models/template.py (hunk 4 above)
Comment (steve-gregory) [inline core/models/template.py]: :mag_right:

File: core/models/status_type.py (hunk 3 above)
Comment (steve-gregory) [inline core/models/status_type.py]: Probably? I'll change this.

File: variables.ini.dist (hunk 5 above)
Comment (lenards) [inline variables.ini.dist]: I'm not sure how this all works when you consider that you can set `TIME_ZONE` on Celery, 

```
# General Celery Settings
CELERY_ENABLE_UTC = True
CELERYD_PREFETCH_MULTIPLIER = 1
CELERY_TIMEZONE = TIME_ZONE
```
source: [atmosphere/settings/__init__.py](https://github.com/iPlantCollaborativeOpenSource/atmosphere/blob/master/atmosphere/settings/__init__.py#L354-L356)

... but I assume without researching in Celery docs, this is all A - :ok:

Comment (steve-gregory) [inline variables.ini.dist]: I do re-set `CELERY_TIMEZONE = TIME_ZONE`, if set:
https://github.com/iPlantCollaborativeOpenSource/atmosphere/pull/127/files#diff-32fcd699d95a8179189b817be4d0aab1R27

Comment (lenards) [inline variables.ini.dist]: :+1: :ok:

**Comment by lenards** [comment]
Unfortunately, this PR did not expose a `api/v2/help_links` endpoint.  Was that an oversight?

**Comment by lenards** [comment]
I _used_ to think I was a good PR reviewer - but in the last week I have amassed evidence suggestions that I am _*not*_. 

Okay, so maybe we just need the entry in `api/v2/urls.py` because I see the view set:
```
 git grep 'help_link'
api/v2/serializers/details/__init__.py:from .help_link import HelpLinkSerializer
api/v2/views/__init__.py:from .help_link import HelpLinkViewSet```
```

**Pull request merged**